### ISSUE TITLE
Have fighters/drone do strafing runs instead of circling their target

### ISSUE THREAD

**Issue by Amazinite** [endless-sky/endless-sky#2220]
This might be a change worth doing after the addition of turret rotational speeds. Say a heavy warship has its turrets almost pointed right at a fighter, and the fighter is not fast enough to be able to circle the warship just ahead of the turret speed. The best thing for the fighter to do in that situation would be to fly over the warship, meaning that its turret is now going to need to do a 180 in order to get a lock back onto the fighter. In essence, this would increase the survivability of fighters and drones, but be balanced off by the reduction in DPS a single fighter is able to do given that it will be spending a lot of time flying past the target then turning around.

I don't know how fast/slow turrets are going to turn and how much it will change the use of fighters and drones, but I feel that this AI change would be especially useful for making fighters and drones more appealing, especially against large and slow targets.

**Comment by KiLEdEnNis** [comment]
wait turrets got rotation speed?

**Comment by jafdy** [comment]
@KiLEdEnNis not yet. It's planed.

**Comment by KiLEdEnNis** [comment]
oh ye, I knew it was planned, ty

**Comment by w0ntoan** [comment]
And then add anti-fighter turrets that are generally weak (e.g. single beam, more weight), turn very fast, and independently targets the nearest hostile fighter? And if there is no fighter within range, aims at selected target?

**Comment by Lineth** [comment]
Maybe something like how Naev handles weapons?

**Comment by Amazinite** [comment]
Explain plz/

**Comment by LocalGod79** [comment]
@Amazinite try equipping larger thrusters (and less capable steering thrusters) - should accomplish the desired effect.

**Comment by Lineth** [comment]
Be able to control what weapon fires at what time by designating them primary or secondary, and being able to set or customize firing groups that save these, so if you have two heavy turrets, two light turrets and two missiles, you can set to fire either the heavies as primaries and the turrets as secondaries and the missiles not at all, or just the heavies, or the heavies and the missiles, etc.

**Comment by Bladewood** [comment]
Reassembly?

**Comment by LocalGod79** [comment]
@Lineth what does your comment have to do with the o.p.?

**Comment by Tadrix** [comment]
Please, give option to specify if the fighter circles or strafes. For _most_ fighters strafing is better, but some gunship types (Barb, turret ver.) are better off circling. That's what real world gunships do. Or hard-code that if the drone/fighter in question has a turret, it circles, if not, it strafes.

**Comment by KiLEdEnNis** [comment]
you mean adding a tag per fighter if they strafe or circle, add an AI tag which the player can't change to the direct hulls?
stop there, just no
just give them bad steering & good thrusters

**Comment by Tadrix** [comment]
> just give them bad steering & good thrusters

This would be strafing.

About tags — that's what the first thought was, but I concluded it would be a bad idea. Slightly better one would be for AI to decide itself, basically, if there is a turret, circles, if isn't — strafes.

**Comment by KiLEdEnNis** [comment]
"just give them bad steering & good thrusters"

> This would be strafing.

that's literally my point: if ya want them to strafe do that
and locking it on turrets etc. isn't a good idea either
if the AI gets updated, make it a preference

**Comment by Tadrix** [comment]
Wait, is this your response to @Amazinite suggestion to make drones and fighters do strafing runs? I can see you reacted with thumbs up to the OP, so your last posts seem weird. Of course, maybe there's a language barrier or something… Only thing I was saying that if they go ahead and change the AI, would be nice to leave old behavior to turreted fighters, as circling around slower target is what gunships actually do.

**Comment by Wrzlprnft** [comment]
@Tadrix, KiLEdEnNiS posts are sometimes a bit confusing, that happens. don't worry, he'll sort it out.

i personally would like the player being able to set wether to strafe or to circle,depending on the fighter and the loadout. if the AI or content creators decide that, it could mess with players plans.

**Comment by KiLEdEnNis** [comment]
@Wrzlprnft ( ͡°, _ʖ ͡°,)

anyway I said that if you want your fighters to strafe you should do the good thrusters bad steering thing, it shouldn't be hard coded in the AI, unless you can set a preference
ya said that would be strafing, which is my point
and even if fighter strafing becomes a preference, ya should be able to edit it per fighter or fighter class, making that messy and unlikely to be added

**Comment by Forlarren** [comment]
KiLEdEnNis would make a good F-4 Phantom II pilot. Who needs to turn when you can just go fast and hit hard? That's a clever use of the engine by the way. I'm going to use that.

**Comment by KiLEdEnNis** [comment]
I think I didn't come up with that but ty anyway, np

**Comment by Pointedstick** [comment]
They do seem to do this now. Large heavy turrets are (appropriately) less useful against swarming ships because of the long rotation time when they fly over you. When playing a solo capital ship, I've recently found myself equipping it with any of the following:
- All heavy turrets, with fast maneuverable missile launchers in the gun ports for shooting down fighters
- All anti-fighter turrets with heavy long-ranged guns or missile launchers of various types in the gun ports
- A few heavy turrets with at least two long-range fast-turning anti-fighter turrets: Quad Blaster turrets, Proton Turrets, Pulse Turrets, Repeater Turrets, and even Thrasher Turrets are excellent for this despite their short range; I really wish the Modified[ Quad Blaster Turret](https://github.com/endless-sky/endless-sky/pull/3319) had been merged because it would have been really good for this as well.

For fleets, I find myself outfitting at least a few ships for anti-fighter duty rather than just cramming the hugest turrets into every ship. Overall I feel like it's been a positive change to make combat more tactically interesting.

In other words, I think this issue can probably be closed. :slightly_smiling_face:

**Comment by Pointedstick** [comment]
This seems to be more or less solved today!

**Comment by Amazinite** [comment]
Popping this one back open because I'm pretty sure it isn't actually the case that this was done.

**Comment by Zitchas** [comment]
It can definitely *feel* like it is done by forcing the issue by giving fighters good engines and weak steering .

On the topic of how anti-fighter weapons worked in other games:
That being said, for an alternate example of how this is handled: in Escape Velocity Nova turrets had an attribute/bit that could be set that was basically a "designate this weapon as an anti-fighter / anti-missile weapon" (EVN lumped anti-missile in with anti-fighter. That does not need to be the case with us.). Any weapon that had that attribute set would basically act like turrets do when they we have opportunistic tracking and auto-fire enabled. Except that they would *only* target fighters and missiles. 

My memory is fuzzy, at this point I don't actually remember if designated anti-fighter turrets were even connected to the manual fire controls at all any more. Either way, in game it results in situations where I would be flying a big ship with the heaviest turrets I could get, pounding away on some big capital ship; and meanwhile my quad light blaster turrets (which were one such anti-fighter weapon) would be swivelling around unleashing a storm of blaster fire trying to knock out any fighter that came within range of me. It was great. 

As far as a desirable target behavior goes, what I'd like to see is a few different things that could work together:
a) a menu as to whether or not anti-fighter weapons work independently, hybrid, or regular weapons. (independently means they always are opportunistic and auto-fire, but will always prioritize drones and fighters over anything else and aren't connected to the main firing controls. Hybrid is they do that, but they can also be force-fired with the regular weapon key. Regular, is as normal now)
b) An "Anti-Fighter" attribute that can be put on any weapon. Weapons with this attribute will act according to the menu in item A.
c) An update to the AI so that fighters and drones can recognize when such weapons are present and change tactics. In EVN, solo fighters would often wait (if backup was likely to show up) to attack all at once. Although lone pirates would often still just charge in and die instantly.
